devlib/module/gem5stats.py

```diff
@@ -66,6 +67,7 @@ def __init__(self, target):
         self._stats_file_path = os.path.join(target.platform.gem5_out_dir,
                                             'stats.txt')
         self.rois = {}
+        self._dump_pos_cache = {0: 0}
```

Comment:
Use a list rather than a dict here? That would make the logic on lines 176-177 simpler (and probably faster).